src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1063,5 +1063,65 @@ public void BlockDataDownloadedIsCalled_ForValidBlocksAfterFv_ResultShouldBeTrue
             resultForH5.Should().BeTrue();
             chainTipH5.Block.Should().NotBeNull();
         }
+
+        /// <summary>
+        /// Issue 26 @  Checkpoints are disabled. Chain tip is at header 5. Reorganisation is initiaised with max reorg of 500 blocks plus extra 50 blocks. 
+        /// We start syncing until block 490. Peer 2 presents the alternative chain with 2 headers after fork point at header 5. We then you join the rest of
+        /// 60 blocks. ConsensusTipChanged should return identifier of the second peer at block number 505.
+        /// </summary>
+        [Fact]
+        public void ChainWithMaxReorgPlusExtraHeadersIsCalled_AnotherChainIsPresented_ConsensusTipChangedReturnsSecondPeerId()
+        {
+            // Chain header tree setup. Initial chain has 5 headers.
+            // Example: h1=h2=h3=h4=h5.
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader initialChainTip = ctx.InitialChainTip;
+
+            // Extend the chain (chain A) with max reorg headers (500) + 50 extra.
+            // Example: h1=h2=h3=h4=(h5)=a6=...=a555.
+            const int maxReorg = 500;
+            ctx.ChainStateMock.Setup(x => x.MaxReorgLength).Returns(maxReorg);
+            ChainedHeader chainATip = ctx.ExtendAChain(maxReorg + 50, initialChainTip);
+
+            // Chain A is presented by peer 1.
+            List<BlockHeader> listOfChainABlockHeaders = ctx.ChainedHeaderToList(chainATip, maxReorg + 50);
+            List<ChainedHeader> listOfChainAChainHeaders = chainATip.ToList(maxReorg + 50);
+            cht.ConnectNewHeaders(1, listOfChainABlockHeaders);
+            
+            // Sync 490 blocks from chain A.
+            for (int i = 0; i < maxReorg - 10; i++)
+            {
+                ChainedHeader currentChainTip = listOfChainAChainHeaders[i];
+                cht.BlockDataDownloaded(currentChainTip, currentChainTip.Block);
+                cht.PartialValidationSucceeded(currentChainTip, out bool reorgRequired);
+                List<int> tips = cht.ConsensusTipChanged(currentChainTip);
```

Comment:
You are not using this list anywhere, but we could actually check that it is null as it should be.